src/Gym.Environments/Envs/Classic/CartPoleEnv.cs

```diff
@@ -107,9 +109,11 @@ public class CartPoleEnv : Env {
                 var pivotPoint = new PointF(cbounds.X + cbounds.Width / 2f, cbounds.Y + cbounds.Height / 2f);
 
                 draw.Add((cart.Translate(center_x, 0), Rgba32.Black));
-                draw.Add((pole.Transform(Matrix3x2.CreateRotation((float) -state.GetDouble(2), pivotPoint)).Translate(center_x, 0), new Rgba32(204, 153, 102)));
+                draw.Add((pole.Transform(Matrix3x2.CreateRotation((float) state.GetDouble(2), pivotPoint))
```

Comment:
When running the game i immediately saw it does not seem to work right. The pole was always moving the same direction of the cart. I went in and forced the env to always push the cart right and the pole always felt right (should fell on the left).

Taking a close look at the code, that's a very weird placed minus right there, isn't it? Why computing the angle "wrong", passing the opposite value to the CreateRotation function and then returning it "wrong" as the Step() function result? I thought the angle was correct but the py function had an issue causing the pole to be rendered with the opposite rotation and they fixed it that way. 

Removing that minus solved the problem (pole acting according to physics) and looked correct to me.

Reply:
Looking at it now, it seems you are right.
It might be the due to a different transformation logic between SixLabors library and Python.
We do need some kind of solid proof because this implies that my computation of the state is wrong which will lead to pre-trained models to fail.